grpc/core/src/main/java/io/helidon/grpc/core/GrpcTlsDescriptor.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2019, 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.grpc.core;
+
+import io.helidon.common.configurable.Resource;
+import io.helidon.config.Config;
+import io.helidon.config.metadata.Configured;
+import io.helidon.config.metadata.ConfiguredOption;
+
+/**
+ * GrpcTlsDescriptor contains details about configuring TLS of a {@link io.grpc.Channel}.
+ */
+public class GrpcTlsDescriptor {
```

Comment:
This class should not be really in `core`, as server connection TLS is configured on webserver, and can never be configured in a different way (I think). 
In the end the same should be valid for client, where we should use the uderlying configuration of the WebClient

Reply:
Removed